### ISSUE TITLE
no-unused-vars warning when variable and type have the same name

### ISSUE THREAD

**Issue by MierenManz** [denoland/deno_lint#705]
Linter thinks type is unused if the variable name is the same as the type name
### Lint Name
`no-unused-vars`

### Code Snippet
a.ts
```ts
import type { Filters } from "./types.ts";
export function audioFilters(...Filters: Filters[]): void {
  for (const filter of Filters) {
    console.log(filter);
    return;
  }
}
```

types.ts
```ts
export interface Filters {
  complex?: boolean;
  filterName: string;
  options: Record<string, string | number>;
}
```
### Expected Result

No warnings

### Actual Result

linter keeps warning that I'm not using `Filters`

### Additional Info

this happens when the type name is the same as the variable name\
using `filters` instead of `Filters` as variable name fixes this for me

### Version

deno 1.10.2 (release, x86_64-unknown-linux-gnu)
v8 9.1.269.27
typescript 4.2.2

**Comment by magurotuna** [comment]
Thanks for the report!

---

Minimal snippet that results in false positive is:

```ts
import type { Foo } from "./foo.ts";
//            ^^^ <---- (a)
function _bar(...Foo: Foo) {
//                    ^^^ <---- (b)
  console.log(Foo);
}

```

But the following code is linted _correctly_ (i.e. true negative):

```ts
type Foo = string[];
//   ^^^ <---- (c)
export function bar(...Foo: Foo): void {
//                          ^^^ <---- (d)
  console.log(Foo);
}
```

Looking into it further, I found that 

- in the former snippet, `Foo` (b) has `ctxt: #2`, while `Foo` (a) has `ctxt: #1`
- in the latter snippet, `Foo` (d) has `ctxt: #1`, which is equivalent to the ctxt of `Foo` (c)

So I think this bug probably arises from the swc's logic of name resolution.
@kdy1 could you investigate it please?

**Comment by kdy1** [comment]
Fixed with a new version of `swc_ecma_transforms_base`.

**Comment by magurotuna** [comment]
@kdy1 Great work, thanks!
